roles/lib_openshift/src/class/oc_project.py

```diff
@@ -0,0 +1,183 @@
+
+        self.project.update_annotation('display-name', self.config.config_options['display_name']['value'])
+        self.project.update_annotation('description', self.config.config_options['description']['value'])
+
+        # work around for immutable project field
+        if self.config.config_options['node_selector']['value']:
+            self.project.update_annotation('node-selector', self.config.config_options['node_selector']['value'])
+        else:
+            self.project.update_annotation('node-selector', self.project.find_annotation('node-selector'))
+
+        return self._replace_content(self.kind, self.config.namespace, self.project.yaml_dict)
+
+    def needs_update(self):
+        ''' verify an update is needed '''
+        result = self.project.find_annotation("display-name")
+        if result != self.config.config_options['display_name']['value']:
+            return True
+
+        result = self.project.find_annotation("description")
+        if result != self.config.config_options['description']['value']:
+            return True
+
+        result = self.project.find_annotation("node-selector")
+        if result != self.config.config_options['node_selector']['value']:
+            return True
+
+        # Check rolebindings and policybindings
+        return False
+
+    # pylint: disable=too-many-return-statements,too-many-branches
+    @staticmethod
+    def run_ansible(params, check_mode):
+        '''run the idempotent ansible code'''
+
+        _ns = None
+        if params['node_selector'] is not None:
+            _ns = ','.join(params['node_selector'])
+
+        pconfig = ProjectConfig(params['name'],
+                                params['name'],
+                                params['kubeconfig'],
+                                {'admin': {'value': params['admin'], 'include': True},
+                                 'admin_role': {'value': params['admin_role'], 'include': True},
+                                 'description': {'value': params['description'], 'include': True},
+                                 'display_name': {'value': params['display_name'], 'include': True},
+                                 'node_selector': {'value': _ns, 'include': True},
+                                })
+
+        oadm_project = OCProject(pconfig, verbose=params['debug'])
+
+        state = params['state']
+
+        api_rval = oadm_project.get()
+
+        #####
+        # Get
+        #####
+        if state == 'list':
+            return {'changed': False, 'results': api_rval['results'], 'state': state}
+
+        ########
+        # Delete
+        ########
+        if state == 'absent':
+            if oadm_project.exists():
+
+                if check_mode:
+                    return {'changed': True, 'msg': 'CHECK_MODE: Would have performed a delete.'}
+
+                api_rval = oadm_project.delete()
+
+                return {'changed': True, 'results': api_rval, 'state': state}
```

Comment:
The response from ``oadm_project.delete()`` isn't being checked to verify it did delete. I recommend checking before noting that ``changed`` is ``True``.